infra/scripts/test-end-to-end-aws.sh

```diff
@@ -1,3 +1,19 @@
 #!/usr/bin/env bash
 
-aws sts get-caller-identity
\ No newline at end of file
+set -euo pipefail
```

Comment:
Where is this file being used?

Reply:
This runs in a k8s cluster in AWS account, kicked off by codebuild job (the configuration for the codebuild job itself is not part of this PR).